Modify/datetime

lpoli opened the pull request:
### Changes
- Optimize allocation changes functions
- Fix issue directory renaming
- Directory can only be created with commit request





### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/blobber/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- gosdk:https://github.com/0chain/gosdk/pull/509

lpoli committed:
Remove whopays attribute

lpoli committed:
Remove dot logger

lpoli committed:
Fix hashes of test

lpoli committed:
Remove redundant payer id field

lpoli committed:
Remove authticket association with readmarker

peterlimg committed:
Merge branch 'staging' into fix/update-attributes

peterlimg committed:
Fix conflicts

lpoli committed:
Remove proto file

lpoli committed:
Merge branch 'fix/update-attributes' of https://github.com/0chain/blobber into fix/update-attributes

peterlimg committed:
Update go modules

peterlimg committed:
Resolve go mod conflicts

lpoli committed:
Merge branch 'staging' into fix/update-attributes

lpoli committed:
Generate/Update grpc code

lpoli committed:
Go mod tidy

lpoli committed:
Remove payer in test insertion

lpoli committed:
Make reader pay for read instead of owner

cnlangzi committed:
Merge branch 'staging' into fix/update-attributes

lpoli committed:
Merge branch 'staging' into fix/update-attributes

lpoli committed:
Go mod tidy

lpoli committed:
error ret

lpoli committed:
Add separate handler for directory creation

lpoli committed:
Separate directory creation handling

lpoli committed:
Replace time.Time with common.Timestamp in refs
Remove deleted_at column
Update tests
Optimize *change.go files

lpoli committed:
Optimize upload file changer

lpoli committed:
Remove unrequired files

lpoli committed:
Remove dot import and fmt prints

lpoli committed:
Merge branch 'staging' into modify/datetime

lpoli committed:
Fix imports

lpoli committed:
Update timestamp

lpoli committed:
Remove calculate hash feature

lpoli committed:
Fix gosimple issue

lpoli committed:
Return nil for root path

lpoli committed:
Reduce slice size

lpoli committed:
Fix rename issues

lpoli committed:
Fix timestamp issue and missing db save

lpoli committed:
Merge branch 'staging' into modify/datetime

service-0chain commented:
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | modify/datetime  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | modify/datetime  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

lpoli committed:
Update timestamp for delete operation

service-0chain commented:
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | modify/datetime  |
| zboxcli  | modify/datetime  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | modify/datetime  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

lpoli committed:
Fix tests

lpoli committed:
Use condition

lpoli committed:
Check error

lpoli committed:
Fix root row retrieval issue

lpoli committed:
Remove deleted columns usages

lpoli committed:
Add log

lpoli committed:
Use deleted_at functionality to delete files
Use children loaded condition

lpoli committed:
Fix missing level and ordering issue

lpoli committed:
Fix missing level issue

service-0chain commented:
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | modify/datetime  |
| gosdk  | modify/datetime  |
| zboxcli  | modify/datetime  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | modify/datetime  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

lpoli committed:
Add path hash

lpoli committed:
Undo path hash removal

lpoli committed:
Uncomment workers and add loop label

service-0chain commented:
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | modify/datetime  |
| gosdk  | modify/datetime  |
| zboxcli  | modify/datetime  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | modify/datetime  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

lpoli committed:
Fix unit tests

lpoli committed:
Merge branch 'staging' into modify/datetime

lpoli committed:
Fix type casting

lpoli committed:
Remove unused function

peterlimg commented on code/go/0chain.net/blobber/worker.go:
```diff
@@ -35,10 +35,11 @@ func setupWorkers(ctx context.Context) {
 		select {
 		case <-ctx.Done():
-			break
+			break loop
```
We can return here instead of using the label

peterlimg commented on code/go/0chain.net/blobber/worker.go:
```diff
@@ -51,10 +52,11 @@ func refreshPriceOnChain(ctx context.Context) {
 		select {
 		case <-ctx.Done():
-			break
+			break loop
```
return

peterlimg commented on code/go/0chain.net/blobber/worker.go:
```diff
@@ -76,10 +78,11 @@ func startRefreshSettings(ctx context.Context) {
 		select {
 		case <-ctx.Done():
-			break
+			break loop
```
return

peterlimg commented on code/go/0chain.net/blobbercore/allocation/newDirchange.go:
```diff
@@ -0,0 +1,121 @@
+package allocation
```
Rename file to newdirchange.go

peterlimg commented on code/go/0chain.net/blobbercore/allocation/newDirchange.go:
```diff
@@ -0,0 +1,121 @@
+func (nf *NewDir) ApplyChange(ctx context.Context, change *AllocationChange,
+	allocationRoot string, ts common.Timestamp) (*reference.Ref, error) {
+
+	fmt.Printf("\n\nApplying change for direcory change\n\n")
```
Use logging please

peterlimg commented on code/go/0chain.net/blobbercore/allocation/newDirchange.go:
```diff
@@ -0,0 +1,121 @@
+}
+
+func (nfch *NewDir) CommitToFileStore(ctx context.Context) error {
+	return nil
```
Not implemented or not needed ?

lpoli replied:
Required for the interface matching but implementation is not required

peterlimg commented on code/go/0chain.net/blobbercore/allocation/newDirchange.go:
```diff
@@ -0,0 +1,121 @@
+	for i := 0; i < len(fields); i++ {
+		found := false
+		for _, child := range dirRef.Children {
+			if child.Name == fields[i] {
```
We need to check if file exists that has the same name as the dir

lpoli replied:
Here we don't need to check because if I get referencePath of path `/a/b/c/d.txt` then `/`, `/a`, `/a/b` and `/a/b/c` are naturally directories.

For `d.txt` we don't need to check because it is checked already if `/a/b/c/d.txt` exists.

peterlimg commented on code/go/0chain.net/blobbercore/allocation/newfilechange.go:
```diff
@@ -1,273 +0,0 @@
-	"github.com/0chain/gosdk/constants"
-)
-
-type NewFileChange struct {
```
Is this NewFileChange be replaced by the UploadFileChanger?

lpoli replied:
Yes

peterlimg commented on code/go/0chain.net/blobbercore/reference/object.go:
```diff
@@ -24,15 +25,22 @@ func DeleteObject(ctx context.Context, allocationID, objPath string) (*Ref, erro
 		for _, ref := range dirRef.Children {
-			if ref.Name == subDir && ref.Type == DIRECTORY {
+			if ref.Name == name {
+				fmt.Printf("\nParent ref: %s\n\n", ref.Path)
```
Use logging or remove if it's for debugging

peterlimg commented on code/go/0chain.net/blobbercore/reference/ref.go:
```diff
@@ -341,38 +350,45 @@ func (fr *Ref) CalculateFileHash(ctx context.Context, saveToDB bool) (string, er
+func (r *Ref) CalculateDirHash(ctx context.Context, saveToDB bool) (h string, err error) {
+	defer func() {
+		if err == nil && saveToDB {
+			err = r.SaveDirRef(ctx)
```
This defer actions would call the r.SaveDirRef when there's no child.

peterlimg commented on code/go/0chain.net/blobbercore/reference/referencepath.go:
```diff
@@ -279,66 +286,66 @@ func GetUpdatedRefs(ctx context.Context, allocationID, path, offsetPath, _type,
-	return
-}
+// //Retrieves deleted refs compared to some update_at value. Useful for localCache.
+// func GetDeletedRefs(ctx context.Context, allocationID, updatedDate, offsetPath, offsetDate string, pageLimit int, dateLayOut string) (refs *[]PaginatedRef, totalPages int, newOffsetPath, newOffsetDate string, err error) {
```
Clean up the commented code if not used

lpoli committed:
Remove commented code

lpoli committed:
Replace loop label with return statement

lpoli committed:
Rename file

lpoli committed:
Remove fmt.Print's

lpoli committed:
Remove fmt.Print 's

lpoli committed:
Skip db update if there is no change

lpoli committed:
Fix unit test

peterlimg reviewed (approved):
lgtm

lpoli committed:
Save newly added directory while root hash calculation

lpoli committed:
Add stats for all newly created directory

lpoli committed:
Fix postgres type cast issue

lpoli committed:
Use HashToBeComputed field whether to calculate hash

lpoli committed:
Fix postgres typecase issue

lpoli committed:
Use step controlling autoincrementor
`autoIncrement` was supposed to work for it fail for sqlite used for integration testing

lpoli committed:
Undo type changes

lpoli committed:
Fix root ref to be considered for hash calculation

lpoli committed:
Merge branch 'staging' into modify/datetime

lpoli committed:
Merge branch 'staging' into modify/datetime

service-0chain commented:
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | TRUE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | modify/datetime  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |
